### PR TITLE
normalize plain text exception extraction on PrettyPageHandler

### PR DESCRIPTION
**Pull request by juliangut**
covers #518

### CODE REVIEW

File: src/Whoops/Resources/views/layout.html.php [hunk 1]
```diff
@@ -3,7 +3,19 @@
...
+<?php echo $preface; ?>
+
+
+
+
+
+
+
+
+
```
Comment (staabm) [inline src/Whoops/Resources/views/layout.html.php]: Why this empty lines?

File: src/Whoops/Handler/PrettyPageHandler.php [hunk 2]
```diff
@@ -202,12 +204,13 @@ public function handle()
             "frame_code"                 => $this->getResource("views/frame_code.html.php"),
             "env_details"                => $this->getResource("views/env_details.html.php"),
 
+            "preface"        => $this->templateHelper->escape($plainTextException),
             "title"          => $this->getPageTitle(),
             "name"           => explode("\\", $inspector->getExceptionName()),
             "message"        => $inspector->getExceptionMessage(),
             "docref_url"     => $inspector->getExceptionDocrefUrl(),
             "code"           => $code,
-            "plain_exception" => Formatter::formatExceptionPlain($inspector),
```
Comment (staabm) [inline src/Whoops/Handler/PrettyPageHandler.php]: Should the method be deprecated then?
Is it used elsewhere?

File: src/Whoops/Handler/PrettyPageHandler.php [hunk 3]
```diff
@@ -238,11 +241,6 @@ public function handle()
         }, $this->getDataTables());
         $vars["tables"] = array_merge($extraTables, $vars["tables"]);
 
-        $plainTextHandler = new PlainTextHandler();
-        $plainTextHandler->setException($this->getException());
-        $plainTextHandler->setInspector($this->getInspector());
-        $vars["preface"] = "<!--\n\n\n" .  $this->templateHelper->escape($plainTextHandler->generateResponse()) . "\n\n\n\n\n\n\n\n\n\n\n-->";
```
Comment (staabm) [inline src/Whoops/Handler/PrettyPageHandler.php]: The html comment lines are missing after your change?

Comment (juliangut) [inline src/Whoops/Handler/PrettyPageHandler.php]: ohh, I forgot to add them on the view  :cold_sweat:

File: src/Whoops/Handler/PrettyPageHandler.php (hunk 2 above)
Comment (juliangut) [inline src/Whoops/Handler/PrettyPageHandler.php]: It can be removed indeed, it's not used anywhere else

Comment (staabm) [inline src/Whoops/Handler/PrettyPageHandler.php]: Not sure we can delete it, but deprecate at least for removal on ntext major version

File: src/Whoops/Resources/views/layout.html.php [hunk 4]
```diff
@@ -3,7 +3,16 @@
 * Layout template file for Whoops's pretty error output.
 */
 ?>
-<!DOCTYPE html><?php echo $preface; ?>
+<!DOCTYPE html>
+
```
Comment (staabm) [inline src/Whoops/Resources/views/layout.html.php]: No newlines before and after the html comment plz

Comment (juliangut) [inline src/Whoops/Resources/views/layout.html.php]: Is this really perceived as a problem?

**Comment by juliangut** [comment]
Is there something missing for this PR to be accepted?? ping @denis-sokolov

**Comment by juliangut** [comment]
Lets see if this PR can now be merged:

* deprecated notice on Formatter::formatExceptionPlain
* removed extra blank lines before and after html comment ;(

**Comment by juliangut** [comment]
ping @denis-sokolov @staabm

File: src/Whoops/Exception/Formatter.php [hunk 5]
```diff
@@ -47,8 +47,20 @@ public static function formatExceptionAsDataArray(Inspector $inspector, $shouldA
...
+     * @param Inspector $inspector
+     *
+     * @return string
+     *
+     * @deprecated
+     */
     public static function formatExceptionPlain(Inspector $inspector)
     {
+        @trigger_error(
+            'Using plain exception formatter is deprecated and will be removed in next releases',
```
Comment (staabm) [inline src/Whoops/Exception/Formatter.php]: The msg should mention what to use instead

Comment (juliangut) [reply src/Whoops/Exception/Formatter.php]: Done, additionally extended the test for deprecation error triggering

**Comment by denis-sokolov** [comment]
Sorry for the delay, and now it led to conflicts with the current master. If you are still interested in this, feel free to revisit, and otherwise my apologies for losing your contribution!

**Pull request closed without merging**